docs/architecture/cluster-registration-workflow.md

```diff
@@ -0,0 +1,40 @@
+# Cluster registration workflow
+
+Registering a cluster in Clusternet is similar to joining a new node to Kubernetes.
+
+The basic registration process is that the agent first sends a registration request to the hub cluster through the bootstrap token of the hub cluster. After the hub cluster detects the registration request, the hub cluster will start to create some necessary resources for registration. The registration request will be marked as approved after these resources to be created and ready.  When the cluster registration success, the agent will periodically send its heartbeat information to the hub cluster.
+
+## Architecture
+
+<img src="../images/cluster-registration-workflow.png" style="width:1000px;"/>
+
+## Workflow
+
+There are 2 core concepts for the cluster registration workflow:
+
+- `Hub` is the control plane of all the registered clusters, run on the parent cluster. It is responsible for approving registration and creating necessary resources for the registered clusters.
```

Comment:
`Hub` is the controller running on the parent cluster to handle cluster registration requests. It is responsible for approving registration requests and creating necessary resources for the registered clusters.